app/src/main/java/org/mozilla/fenix/components/metrics/AppLaunchTimeMeasurement.kt

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.os.Process
+import android.os.SystemClock
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.mozilla.fenix.perf.Stat
+
+/**
+ * Handles the logic of figuring out launch time for cold, warm and hot startup.
+ */
+class AppLaunchTimeMeasurement(private val stats: Stat = Stat()) {
+    private var isColdStartup = false
+    private var isWarmStartup = false
+    private var isHotStartup = false
```

Comment:
This is acceptable – no action needed unless you want to – but consider a red flag here: these states are mutually exclusive. We're using 3 booleans to represent one state value. A safer formation is:
```kotlin
// can we re-use this enum from somewhere else without breaking
// encapsulation?
enum class StartupMode { COLD, WARM, HOT, UNKNOWN }

private var mode = StartupMode.UNKNOWN

fun ...onCreate() {
    mode = StartupMode.COLD
}
```

This makes it more intuitive to the programmer when invalid states can happen, e.g. we can't get `isColdStartup == true && isWarmStartup == true`. And we can do fairly straightforward validation on the `mode` property – i.e. only UNKNOWN -> COLD/WARM/HOT or COLD/WARM/HOT -> UNKNOWN – rather than checking 3 booleans.

Reply:
moved it to the enum convention!